src/main/webapp/scripts/tests/controller-auth-test.js

```diff
@@ -0,0 +1,111 @@
+import * as view from '/scripts/view.js';
+import * as controller from '/scripts/controller-auth.js';
+
+// Test (controller-auth.js and) view.js
+QUnit.module('Authorization controller', {
+  beforeEach: () => {
+    gapi = new CustomGapi();
+  },
+  afterEach: () => {
+    sinon.restore();
+  }
+});
+
+QUnit.test('When init fails, displays the authorization button and the error message', async function (assert) {
+  const errorMsg = 'error message';
+  sinon.stub(gapi.client, 'init').returns(Promise.reject(new Error(errorMsg)));
+
+  await controller.initClient();
+
+  assert.dom('#result-text').hasProperty('innerText', 'Failed to initialize client');
+});
+
+QUnit.test('When signed in, there should be sign-out and comment buttons', async function (assert) {
+  sinon.stub(gapi.client, 'init').returns(Promise.resolve());
+  sinon.stub(gapi.auth2, 'getAuthInstance').returns(
+    {
+      'isSignedIn':
+      {
+        'get': () => { return true; },
+        'listen': () => { },
+      }
+    });
```

Comment:
Put this into a helper method that accepts the isSignedIn boolean as a parameter and call it in your tests.

Reply:
Done.